projects/js-packages/publicize-components/src/components/share-limits-bar/index.tsx

```diff
@@ -1,53 +1,115 @@
 import { RecordMeterBar, Text } from '@automattic/jetpack-components';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
 import { useMemo } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import { ShareLimits } from '../../hooks/use-share-limits';
 import styles from './styles.module.scss';
 
+type NoticeType = ShareLimits[ 'noticeType' ];
+
 export type ShareLimitsBarProps = {
-	currentCount: number;
+	usedCount: number;
 	scheduledCount: number;
 	enabledConnectionsCount?: number;
-	maxCount: number;
+	remainingCount?: number;
+	limit: number;
 	text?: string;
+	textVariant?: string;
 	className?: string;
+	noticeType?: NoticeType;
+};
+
+const colorsForUsed: Record< NoticeType, string > = {
+	error: 'var(--jp-red-50)',
+	warning: 'var(--jp-yellow-20)',
+	default: 'var(--jp-green)',
+};
+
+const colorsForScheduled: Record< NoticeType, string > = {
+	error: 'var(--jp-red-50)',
+	warning: 'var(--jp-yellow-20)',
+	default: 'var(--jp-green-5)',
 };
 
 export const ShareLimitsBar = ( {
-	currentCount,
-	maxCount,
+	usedCount,
+	limit,
 	scheduledCount,
-	enabledConnectionsCount,
+	enabledConnectionsCount = 0,
+	remainingCount,
 	text,
+	textVariant = 'body',
 	className,
+	noticeType = 'default',
 }: ShareLimitsBarProps ) => {
+	const { isEditedPostBeingScheduled } = useSelect( editorStore, [] );
+	const isScheduled = isEditedPostBeingScheduled();
+
+	const remaining = Math.max(
+		remainingCount ?? limit - usedCount - scheduledCount - enabledConnectionsCount,
+		0
+	);
+
 	const items = useMemo( () => {
+		let widthConsumed = 0;
+
 		return [
 			{
-				count: currentCount,
-				backgroundColor: 'var(--jp-gray-90)',
-				label: __( 'Shares used', 'jetpack' ),
+				count: usedCount + ( ! isScheduled ? enabledConnectionsCount : 0 ),
+				backgroundColor: colorsForUsed[ noticeType ],
+				label: __( 'used', 'jetpack' ),
 			},
-			enabledConnectionsCount !== undefined && {
-				count: enabledConnectionsCount,
-				backgroundColor: 'var(--jp-gray-40)',
-				label: __( 'Enabled connections', 'jetpack' ),
+			{
+				count: scheduledCount + ( isScheduled ? enabledConnectionsCount : 0 ),
+				backgroundColor: colorsForScheduled[ noticeType ],
+				label: __( 'scheduled', 'jetpack' ),
 			},
 			{
-				count: scheduledCount,
-				backgroundColor: 'var(--jp-gray-20)',
-				label: __( 'Shares scheduled', 'jetpack' ),
+				count: remaining,
+				backgroundColor: 'var(--jp-gray-off)',
+				label: __( 'left', 'jetpack' ),
```

Comment:
`left` may be confusing for translators. May be we should add a comment for them that it means "remaining" not the direction "left/right".